homeassistant/config_entries.py

```diff
@@ -736,9 +750,24 @@ def async_create_task(
         target: target to call.
         """
         task = hass.async_create_task(target)
+        self._tasks.add(task)
+        task.add_done_callback(self._tasks.remove)
+
+        return task
+
+    @callback
+    def async_create_background_task(
+        self, hass: HomeAssistant, target: Coroutine[Any, Any, _R], name: str
```

Comment:
I hate that we have to pass `hass` to the config entry functions. However, I followed the pattern as it's weird if 1 has no `hass` and others do? Although `async_create_task` is fairly new here too, so we could adjust both.